### PR TITLE
Make MapLoader support qrc sources

### PR DESCRIPTION
**Pull request by mitchcurtis**

### CODE REVIEW

**Comment by mitchcurtis** [comment]
The docs for [`isLocalFile`](http://doc.qt.io/qt-5/qurl.html#isLocalFile) say:

> A URL is a local file path if the scheme is "file"

File: src/tiledquickplugin/maploader.cpp [hunk 1]
```diff
@@ -46,7 +46,9 @@ void MapLoader::setSource(const QUrl &source)
 
     Tiled::MapReader mapReader;
 
-    Tiled::Map *map = mapReader.readMap(source.toLocalFile());
+    const QString qrcScheme = QStringLiteral("qrc");
+    const bool isQrcScheme = source.scheme() == qrcScheme;
+    Tiled::Map *map = mapReader.readMap(isQrcScheme ? source.toString().mid(qrcScheme.length()) : source.toLocalFile());
```
Comment (bjorn) [inline src/tiledquickplugin/maploader.cpp]: I don't understand. If the file to be read is a `qrc` file, you're taking the `qrc` prefix out? Why not just keep it? Or maybe you could try `source.path()` instead?

Comment (mitchcurtis) [inline src/tiledquickplugin/maploader.cpp]: My experience has been that you generally only use the qrc scheme in QML - when it comes to C++, it's always ":/my/path". That's probably a simplified understanding of the problem, but it is supported by e.g. [this code](url). If you think the way I've done it is weird, I could copy that code.

Comment (bjorn) [inline src/tiledquickplugin/maploader.cpp]: Right, I'd include a copy of `urlToLocalFileOrQrc` then and use that. It's unfortunate the QQmlFile class is `\internal`, so we shouldn't use it directly even though it's in public headers.

**Pull request merged**